File: enve/Plugins/ProviderSyncStrategy.swift
import Foundation

struct ProviderSyncResult {
    let pulled: Int
    let pushed: Int

    static let zero = ProviderSyncResult(pulled: 0, pushed: 0)
}

@MainActor
protocol ProviderSyncStrategy: AnyObject {

    var id: String { get }

    var displayName: String { get }

    func sync(force: Bool, launchOptimized: Bool) async -> ProviderSyncResult
}

File: android/core/src/main/java/com/enve/core/data/sync/ProviderSyncStrategy.kt
package com.enve.core.data.sync

data class ProviderSyncResult(
    val pulled: Int,
    val pushed: Int,
    val failedBackends: List<String> = emptyList(),
) {
    companion object {
        val ZERO = ProviderSyncResult(pulled = 0, pushed = 0)
    }
}

interface ProviderSyncStrategy {

    val id: String

    val displayName: String

    suspend fun sync(force: Boolean, launchOptimized: Boolean): ProviderSyncResult
}
